resources/js/features/game-list/components/AllGamesDataTable/buildColumnDefinitions.tsx

```diff
@@ -0,0 +1,44 @@
+import type { ColumnDef } from '@tanstack/react-table';
+
+import { buildAchievementsPublishedColumnDef } from '../../utils/column-definitions/buildAchievementsPublishedColumnDef';
+import { buildLastUpdatedColumnDef } from '../../utils/column-definitions/buildLastUpdatedColumnDef';
+import { buildNumUnresolvedTicketsColumnDef } from '../../utils/column-definitions/buildNumUnresolvedTicketsColumnDef';
+import { buildNumVisibleLeaderboardsColumnDef } from '../../utils/column-definitions/buildNumVisibleLeaderboardsColumnDef';
+import { buildPlayerGameProgressColumnDef } from '../../utils/column-definitions/buildPlayerGameProgressColumnDef';
+import { buildPlayersTotalColumnDef } from '../../utils/column-definitions/buildPlayersTotalColumnDef';
+import { buildPointsTotalColumnDef } from '../../utils/column-definitions/buildPointsTotalColumnDef';
+import { buildReleasedAtColumnDef } from '../../utils/column-definitions/buildReleasedAtColumnDef';
+import { buildRetroRatioColumnDef } from '../../utils/column-definitions/buildRetroRatioColumnDef';
+import { buildRowActionsColumnDef } from '../../utils/column-definitions/buildRowActionsColumnDef';
+import { buildSystemColumnDef } from '../../utils/column-definitions/buildSystemColumnDef';
+import { buildTitleColumnDef } from '../../utils/column-definitions/buildTitleColumnDef';
+
+export function buildColumnDefinitions(options: {
+  canSeeOpenTicketsColumn: boolean;
+  forUsername?: string;
+}): ColumnDef<App.Platform.Data.GameListEntry>[] {
+  const columnDefinitions: ColumnDef<App.Platform.Data.GameListEntry>[] = [
+    buildTitleColumnDef({ forUsername: options.forUsername }),
+    buildSystemColumnDef({}),
+    buildAchievementsPublishedColumnDef({}),
+    buildPointsTotalColumnDef({}),
+    buildRetroRatioColumnDef({}),
+    buildLastUpdatedColumnDef({}),
+    buildReleasedAtColumnDef({}),
+    buildPlayersTotalColumnDef({}),
+    buildNumVisibleLeaderboardsColumnDef({}),
```

Comment:
Column definitions are now reusable functions.

It's plausible some datatables may have columns that aren't used by other tables. By keeping these as individual functions, Vite can perform dead code elimination for the user's page, meaning they only receive the JS that's actually needed for where they're at.